### PR TITLE
bump actions/checkout v4

### PR DESCRIPTION
**Pull request by shogo82148**
I will fix the following warning messages.

> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

e.g. https://github.com/valkey-io/valkey/actions/runs/8482578610

image

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,15 +71,17 @@ jobs:
...
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: make
       run: make SERVER_CFLAGS='-Werror' MALLOC=libc
 
   build-centos7-jemalloc:
     runs-on: ubuntu-latest
     container: centos:7
     steps:
+    # on centos7, actions/checkout@v4 does not work, so we use v3
```
Comment (madolson) [inline .github/workflows/ci.yml:82]: Do you know why this doesn't work? Is there some other issue tracking this that we can post here and/or is it something we can fix now.

Comment (shogo82148) [reply .github/workflows/ci.yml:82]: My execution log is here: https://github.com/shogo82148/valkey/actions/runs/8483453918/job/23244603489

```
 /usr/bin/docker exec  4aa8060332d35ce88381371e8a04932f552e925f8fd84e2e7713525cb970c387 sh -c "cat /etc/*release | grep ^ID"
/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /__e/node20/bin/node)
```

It looks that the version of GLIBC is mismatch.

Comment (madolson) [reply .github/workflows/ci.yml:82]: Seems like this https://github.com/actions/checkout/issues/1487

Comment (shogo82148) [reply .github/workflows/ci.yml:82]: Thanks, I added a link for reference.
https://github.com/valkey-io/valkey/pull/87/commits/01e400958f4ee64ffb6761a9354629fb0bfa19e5

**Review by madolson — APPROVED** [review]

**Pull request merged**